tests/codegen.rs

```diff
@@ -1,118 +1,71 @@
 use std::ascii::escape_default;
-use std::collections::HashMap;
+use std::collections::HashSet;
 use std::fmt::Write;
 use std::fs;
-use std::io::Cursor;
 
+use chrono::{NaiveDate, Utc};
+use num_bigint::BigUint;
 use ring::digest;
+use serde::Deserialize;
+use x509_parser::prelude::AttributeTypeAndValue;
+use x509_parser::x509::X509Name;
 
 #[tokio::test]
-async fn generated_code_is_fresh() {
-    // Fetch the list of certificates as a PEM file from mkcert.org
+async fn new_generated_code_is_fresh() {
+    // Configure a Reqwest client that only trusts the CA certificate expected to be the
```

Comment:
How often does this expire? Should we have some context here on the process used to refresh this cert? Always sucks when CI starts failing because some cert has expired...

Reply:
Trust anchor expiry shouldn't be an issue (recall webpki's representation doesn't even hold notbefore/notafter).

I think the most likely reason this will cause CI failures is if the upstream Salesforce server that backs the CCADB reports changes its TLS configuration. For example switching to a different CA entirely.

I'm open to dropping this pinning if folks think the hassle is greater than the minor security win.

> Should we have some context here on the process used to refresh this cert? Always sucks when CI starts failing because some cert has expired...

Sure, maybe that's a better compromise vs removing it entirely. I'll give this some thought (_I'll pick this up tomorrow, just about to go AFK_).

Reply:
> I'm open to dropping this pinning if folks think the hassle is greater than the minor security win.

I think the pinning is probably nice given the sensitivity here, just want to get out in front of the "CI fails not because of a change I made and now I need to dig a hole to figure out how I fix it up again".

Reply:
I'd like to keep this pinning. I don't see salesforce switching CAs, but it's perhaps more likely they give up hosting this stuff and we have to adjust several things at that time.

Reply:
Honestly, hosting these artifacts at `ccadb.my.salesforce-sites.com/mozilla` feels like an awkward choice for something like the CCADB.

Reply:
It's a really unfortunate choice. The [licensing situation](https://groups.google.com/a/ccadb.org/g/public/c/JlsgHTI0Apg/m/Gi_5QDOMCQAJ?utm_medium=email&utm_source=footer) for login access is not good and artificial limitations force [terrible workarounds](https://groups.google.com/a/ccadb.org/g/public/c/CKoJEAO6Qho/m/AUwnEYLnBAAJ?utm_medium=email&utm_source=footer) requiring evaluating javascript (!). </rant>